### PR TITLE
fix(build): normalize Windows backslash paths for sourcemap resolution

### PR DESCRIPTION
**Pull request by BYK**
## Summary

- Normalize Windows backslashes (`\`) to forward slashes (`/`) in the telemetry `beforeSend` hook so frame paths match uploaded sourcemap artifact URLs
- Normalize the sourcemap `sources` array during debug ID injection so bundlers on Windows produce consistent paths

Fixes https://sentry.sentry.io/issues/7402821077/

## Problem

On Windows, Bun compiled binaries resolve `Error.stack` frame paths with backslashes (e.g., `dist-bin\bin.js`). The `ensureAbsolute()` function in `telemetry.ts` prepended `/` but didn't convert backslashes, producing `/dist-bin\bin.js`. The Sentry symbolicator couldn't match this to the uploaded artifact at `~/dist-bin/bin.js`, so sourcemaps were never applied — stack traces showed minified function names like `ut` with no source context.

## Changes

- **`src/lib/telemetry.ts`**: Add `.replaceAll("\\", "/")` in `ensureAbsolute()` before the relative-path check. Covers `frame.filename`, `frame.abs_path`, and `debug_meta.images[].code_file`.
- **`src/lib/sourcemap/debug-id.ts`**: Normalize `sources` array to forward slashes after parsing the sourcemap JSON. Benefits both the CLI's own build and user-facing `sentry sourcemap inject`.
- **`test/script/debug-id.test.ts`**: Two new test cases for backslash normalization in `sources` and forward-slash preservation.

### CODE REVIEW

File: src/lib/sourcemap/debug-id.ts [hunk 1]
```diff
@@ -133,9 +133,19 @@ export async function injectDebugId(
...
     debug_id?: string;
     debugId?: string;
   };
+
+  // Normalize Windows backslashes in the sources array so uploaded
+  // sourcemaps have consistent forward-slash paths regardless of build
+  // platform. Bundlers on Windows (esbuild, Bun) may produce paths like
+  // "src\\bin.ts". No-op on Linux/macOS.
+  if (map.sources) {
+    map.sources = map.sources.map((s) => s.replaceAll("\\", "/"));
```
[the remark this answers is not in the document]
Comment (BYK) [reply src/lib/sourcemap/debug-id.ts]: Good catch — Source Map v3 spec allows `null` entries in `sources`. Fixed in 70ea7ef6: added a null guard (`s ? s.replaceAll(...) : s`) and a test case for this edge case.

[the remark this answers is not in the document]
Comment (BYK) [reply src/lib/sourcemap/debug-id.ts]: Valid — fixed in 70ea7ef6 with a null guard and test case.
